src/pretix/base/services/cart.py

```diff
@@ -286,11 +291,22 @@ def __init__(self, event: Event, cart_id: str, sales_channel: SalesChannel,
         self._subevents_cache = {}
         self._variations_cache = {}
         self._seated_cache = {}
-        self._expiry = None
-        self._explicit_expiry = expiry
         self.invoice_address = invoice_address
         self._widget_data = widget_data or {}
         self._sales_channel = sales_channel
+        self.num_extended_positions = 0
+
+        if expiry and reservation_time:
+            raise TypeError('Cannot specify both expiry and reservation_time')
+        elif expiry:
+            warnings.warn('CartManager(expiry=...) is deprecated, use reservation_time=... instead')
```

Comment:
if we keep this (see slack), we can properly classify it:

```suggestion
            warnings.warn('CartManager(expiry=...) is deprecated, use reservation_time=... instead', DeprecationWarning)
```

Reply:
Removed the argument altogether